cluster/addons/iso-client/coreaffinity-isolator.go

```diff
@@ -24,37 +27,73 @@ const (
 	name = "iso"
 )
 
+func handleSIGTERM(sigterm chan os.Signal, client *aff.EventDispatcherClient, opaque *opaq.OpaqueIntegerResourceAdvertiser) {
+	<-sigterm
+	shutdownIsolator(client, opaque)
+}
+
+func shutdownIsolator(client *aff.EventDispatcherClient, opaque *opaq.OpaqueIntegerResourceAdvertiser) {
+	unregisterRequest := &lifecycle.UnregisterRequest{
+		Name:  client.Name,
+		Token: client.Token,
+	}
+
+	if _, err := client.Unregister(client.Ctx, unregisterRequest); err != nil {
+		opaque.RemoveOpaqueResource()
+		glog.Fatalf("Failed to unregister handler: %v")
+	}
+	glog.Infof("Unregistering custom-isolator: %s", name)
+
+	if err := opaque.RemoveOpaqueResource(); err != nil {
```

Comment:
Could we log before attempting to remove the opaque resource? (e.g. "Removing opaque resource %s from node %s")